lib/buffer.js

```diff
@@ -54,17 +54,19 @@ function alignPool() {
   }
 }
 
+const bufferDeprecationWarning =
+    deprecate(() => {}, 'Buffer constructor is deprecated. ' +
+      'Use Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.');
+
 /**
- * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
- * in the documentation and should not be used moving forward. Rather,
- * developers should use one of the three new factory APIs: Buffer.from(),
- * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
- * is no hard deprecation because of the extent to which the Buffer constructor
- * is used in the ecosystem currently -- a hard deprecation would introduce too
- * much breakage at this time. It's not likely that the Buffer constructors
- * would ever actually be removed.
+ * The Buffer() construtor is deprecated ... that is, it should not be used
+ * moving forward. Rather, developers should use one of the three new factory
+ * APIs: Buffer.from(), Buffer.allocUnsafe() or Buffer.alloc() based on their
+ * specific needs. It's not likely that the Buffer constructors would ever
+ * actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  bufferDeprecationWarning();
```

Comment:
We could, but I don't think we should.

* If we hard-deprecate `new Buffer(string, encoding)` now, we'll have an option to remove it later. As far as I know, implementing it with the same behavior in `class Buffer` is quite complicated, if possible at all.
- `new Buffer(int)` needs to stay for compatibility with Uint8Array, but hard-deprecating it gives us leeway to change its behavior. Currently, `new Buffer(int)` returns non-zeroed and pooled memory (fast), but `new Uint8Array(int)` returns zeroed and non-pooled memory (slow). If `new Buffer(int)` is hard-deprecated for 6 months, it's likely that much fewer people would be affected by the sudden performance change.

Again, hard-deprecation doesn't necessarily mean removing it later, but it at least allows us to consider it.

And in any case, the Buffer constructor is already soft-deprecated, which itself is a reason for hard-deprecation (see the first point in OP).